# Livro para consulta:
- https://jakevdp.github.io/PythonDataScienceHandbook/03.08-aggregation-and-grouping.html
- https://jakevdp.github.io/PythonDataScienceHandbook/03.09-pivot-tables.html
    

# 1. Importando bibliotecas <a name="import"></a>

<div style="text-align: right"
     
[Voltar ao índice](#Contents)

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

%matplotlib inline

# 2. Carregando o dataframe SINASC <a name="read"></a>
<div style="text-align: right"
     
[Voltar ao índice](#Contents)

In [2]:
sinasc_raw = pd.read_csv('SINASC_RO_2019.csv')
sinasc_raw.head()

ORIGEM   CODESTAB  CODMUNNASC  LOCNASC  IDADEMAE  ESTCIVMAE  \
0       1  2679477.0      110001        1        19        5.0   
1       1  2679477.0      110001        1        29        2.0   
2       1  2679477.0      110001        1        37        9.0   
3       1  2516500.0      110001        1        30        5.0   
4       1  2516500.0      110001        1        30        2.0   

            ESCMAE  CODOCUPMAE  QTDFILVIVO  QTDFILMORT  ...  KOTELCHUCK  \
0      8 a 11 anos         NaN         0.0         0.0  ...           5   
1      8 a 11 anos    999992.0         1.0         0.0  ...           5   
2      8 a 11 anos    513205.0         2.0         0.0  ...           5   
3  12 anos ou mais    231205.0         0.0         0.0  ...           4   
4      8 a 11 anos    999992.0         1.0         0.0  ...           5   

  CONTADOR munResStatus munResTipo               munResNome  munResUf  \
0        1        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
1        2        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
2        3        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
3        4        ATIVO      MUNIC  Alto Alegre dos Parecis  Rondônia   
4        5        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   

   munResLat munResLon  munResAlt  munResArea  
0  -11.93554 -61.99982      338.0    7067.025  
1  -11.93554 -61.99982      338.0    7067.025  
2  -11.93554 -61.99982      338.0    7067.025  
3  -12.13178 -61.85308      397.0    3958.273  
4  -11.93554 -61.99982      338.0    7067.025  

[5 rows x 69 columns]

# Tarefa 1

### 1. Idade media das mães e dos pais por município (coluna munResNome)


In [6]:
# Idade média das mães e dos pais por município
idade_maes_pais_municipio = sinasc_raw.groupby('munResNome')[['IDADEMAE', 'IDADEPAI']].mean().reset_index()

# Exibindo os resultados
print("Idade média das mães e dos pais por município:")
print(idade_maes_pais_municipio)

Idade média das mães e dos pais por município:
                   munResNome   IDADEMAE   IDADEPAI
0       Alta Floresta D'Oeste  25.991826  29.548701
1     Alto Alegre dos Parecis  24.844156  29.179104
2                Alto Paraíso  24.960177  28.833333
3            Alvorada D'Oeste  25.770732  30.757282
4                   Ariquemes  25.607866  32.466667
5                     Buritis  25.559177  30.812500
6                      Cabixi  26.037500  34.333333
7                 Cacaulândia  25.546667  36.333333
8                      Cacoal  26.890102  30.931854
9      Campo Novo de Rondônia  24.825301  30.500000
10         Candeias do Jamari  25.243323  29.894737
11               Castanheiras  27.281250  30.392857
12                 Cerejeiras  27.240506  31.652778
13                Chupinguaia  25.333333  29.629630
14          Colorado do Oeste  27.602804  34.354839
15                 Corumbiara  24.716535  32.470588
16              Costa Marques  24.412017  30.613636
17               

### 2. Peso médio dos bebes por sexo que nasceram no dia do seu aniversário por faixas de escolaridade mae
Ex: Você, aluna(o), nasceu no dia 10/01, então você precisa filtrar o conjunto de dados nessa data e calcular o peso médio dos bebês de cada sexo por faixa de escolaridade da mãe.

In [10]:
# Data do seu aniversário
data_aniversario = '2019-10-05'  # Substitua pelo seu aniversário

# Filtrando os bebês nascidos nessa data
bebes_aniversario = sinasc_raw[sinasc_raw['DTNASC'] == data_aniversario]

# Peso médio dos bebês por sexo e escolaridade da mãe
peso_medio_bebes = bebes_aniversario.groupby(['SEXO', 'ESCMAE'])['PESO'].mean().reset_index()

# Exibindo os resultados
print(f"Peso médio dos bebês nascidos no seu aniversário ({data_aniversario}) por sexo e escolaridade da mãe:")
print(peso_medio_bebes)

Peso médio dos bebês nascidos no seu aniversário (2019-10-05) por sexo e escolaridade da mãe:
        SEXO           ESCMAE         PESO
0   Feminino       1 a 3 anos  2647.500000
1   Feminino  12 anos ou mais  3157.500000
2   Feminino       4 a 7 anos  3417.500000
3   Feminino      8 a 11 anos  3259.736842
4  Masculino       1 a 3 anos  3240.000000
5  Masculino  12 anos ou mais  3392.500000
6  Masculino       4 a 7 anos  3340.000000
7  Masculino      8 a 11 anos  3397.173913


### 3. Qual o municipio que nasceu menos bebe em 2019?
    - qual a idade media, maxima, minima das maes nesse municipio?
    - qual a idade media, maxima, minima dos pais nesse municipio?

In [8]:
# Contando nascimentos por município
nascimentos_por_municipio = sinasc_raw['munResNome'].value_counts()

# Município com o menor número de nascimentos
municipio_menor_nascimento = nascimentos_por_municipio.idxmin()

# Filtrando os dados do município
dados_menor_municipio = sinasc_raw[sinasc_raw['munResNome'] == municipio_menor_nascimento]

# Estatísticas de idade das mães e pais
idade_maes_menor = dados_menor_municipio['IDADEMAE'].agg(['mean', 'min', 'max'])
idade_pais_menor = dados_menor_municipio['IDADEPAI'].agg(['mean', 'min', 'max'])

# Exibindo os resultados
print(f"Município com o menor número de nascimentos em 2019: {municipio_menor_nascimento}")
print(f"Estatísticas de idade das mães nesse município:\n{idade_maes_menor}")
print(f"Estatísticas de idade dos pais nesse município:\n{idade_pais_menor}")

Município com o menor número de nascimentos em 2019: Município ignorado - RO
Estatísticas de idade das mães nesse município:
mean    24.0
min     24.0
max     24.0
Name: IDADEMAE, dtype: float64
Estatísticas de idade dos pais nesse município:
mean    22.0
min     22.0
max     22.0
Name: IDADEPAI, dtype: float64


### 4. Qual o municipio que nasceu mais bebe no mês de março?
    - qual a quantidade de filhos vivos media, maxima, minima nesse municipio?
    - qual a idade media, maxima, minima dos pais nesse municipio?



In [9]:
# Filtrando nascimentos em março
nascimentos_marco = sinasc_raw[sinasc_raw['DTNASC'].str.contains('-03-')]

# Contando nascimentos por município
municipio_maior_marco = nascimentos_marco['munResNome'].value_counts().idxmax()

# Filtrando dados do município
dados_maior_marco = nascimentos_marco[nascimentos_marco['munResNome'] == municipio_maior_marco]

# Estatísticas de quantidade de filhos vivos
filhos_vivos_marco = dados_maior_marco['QTDFILVIVO'].agg(['mean', 'min', 'max'])

# Estatísticas de idade dos pais
idade_pais_maior = dados_maior_marco['IDADEPAI'].agg(['mean', 'min', 'max'])

# Exibindo os resultados
print(f"Município com o maior número de nascimentos em março: {municipio_maior_marco}")
print(f"Estatísticas de quantidade de filhos vivos nesse município:\n{filhos_vivos_marco}")
print(f"Estatísticas de idade dos pais nesse município:\n{idade_pais_maior}")

Município com o maior número de nascimentos em março: Porto Velho
Estatísticas de quantidade de filhos vivos nesse município:
mean    1.295056
min     0.000000
max     9.000000
Name: QTDFILVIVO, dtype: float64
Estatísticas de idade dos pais nesse município:
mean    34.629032
min     19.000000
max     62.000000
Name: IDADEPAI, dtype: float64


### Analise as respostas encontradas, tire algum insight delas, conte pra gente algo encontrado nos dados. Algo que você julgue relevante e novo pra você.

Exemplo:
- Ah, descobri que a idade mediana das mulheres que deram a luz no ano de 2019 dos municipios x é maior que y.

In [3]:
sinasc_raw.dtypes

ORIGEM          int64
CODESTAB      float64
CODMUNNASC      int64
LOCNASC         int64
IDADEMAE        int64
               ...   
munResUf       object
munResLat     float64
munResLon     float64
munResAlt     float64
munResArea    float64
Length: 69, dtype: object

In [4]:
sinasc_raw.select_dtypes(exclude=['int64','float64'])

ESCMAE         GESTACAO GRAVIDEZ    PARTO      DTNASC  \
0          8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-02-19   
1          8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-02-21   
2          8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-02-25   
3      12 anos ou mais  37 a 41 semanas    Única  Cesáreo  2019-03-20   
4          8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-03-23   
...                ...              ...      ...      ...         ...   
27023  12 anos ou mais  32 a 36 semanas    Única  Cesáreo  2019-12-13   
27024      8 a 11 anos  37 a 41 semanas    Única  Vaginal  2019-10-04   
27025      8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-08-02   
27026      8 a 11 anos  32 a 36 semanas    Única  Vaginal  2019-12-23   
27027      8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-12-27   

            SEXO RACACOR  DTCADASTRO CODANOMAL VERSAOSIST   DTRECEBIM  \
0      Masculino  Branca  2019-03-08       NaN     3.2.01  2019-03-08   
1       Feminino  Branca  2019-03-08       NaN     3.2.01  2019-03-08   
2       Feminino  Branca  2019-03-08       NaN     3.2.01  2019-03-08   
3       Feminino   Parda  2019-04-03       NaN     3.2.01  2019-04-09   
4       Feminino   Parda  2019-04-03       NaN     3.2.01  2019-04-09   
...          ...     ...         ...       ...        ...         ...   
27023  Masculino  Branca  2020-09-28       NaN     3.2.01  2020-10-13   
27024  Masculino  Branca  2019-10-16       NaN     3.2.01  2019-10-21   
27025  Masculino   Parda  2019-08-07       NaN     3.2.00  2019-08-07   
27026  Masculino   Parda  2020-01-07       NaN     3.2.00  2020-01-07   
27027  Masculino   Parda  2020-10-15       NaN     3.2.01  2020-10-16   

      munResStatus munResTipo               munResNome  munResUf  
0            ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia  
1            ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia  
2            ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia  
3            ATIVO      MUNIC  Alto Alegre dos Parecis  Rondônia  
4            ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia  
...            ...        ...                      ...       ...  
27023        ATIVO      MUNIC                  Vilhena  Rondônia  
27024        ATIVO      MUNIC              Chupinguaia  Rondônia  
27025        ATIVO      MUNIC                  Vilhena  Rondônia  
27026        ATIVO      MUNIC                  Vilhena  Rondônia  
27027        ATIVO      MUNIC  Novo Horizonte do Oeste  Rondônia  

[27028 rows x 15 columns]

In [11]:
sinasc_raw.columns

Index(['ORIGEM', 'CODESTAB', 'CODMUNNASC', 'LOCNASC', 'IDADEMAE', 'ESTCIVMAE',
       'ESCMAE', 'CODOCUPMAE', 'QTDFILVIVO', 'QTDFILMORT', 'CODMUNRES',
       'GESTACAO', 'GRAVIDEZ', 'PARTO', 'CONSULTAS', 'DTNASC', 'HORANASC',
       'SEXO', 'APGAR1', 'APGAR5', 'RACACOR', 'PESO', 'IDANOMAL', 'DTCADASTRO',
       'CODANOMAL', 'NUMEROLOTE', 'VERSAOSIST', 'DTRECEBIM', 'DIFDATA',
       'DTRECORIGA', 'NATURALMAE', 'CODMUNNATU', 'CODUFNATU', 'ESCMAE2010',
       'SERIESCMAE', 'DTNASCMAE', 'RACACORMAE', 'QTDGESTANT', 'QTDPARTNOR',
       'QTDPARTCES', 'IDADEPAI', 'DTULTMENST', 'SEMAGESTAC', 'TPMETESTIM',
       'CONSPRENAT', 'MESPRENAT', 'TPAPRESENT', 'STTRABPART', 'STCESPARTO',
       'TPNASCASSI', 'TPFUNCRESP', 'TPDOCRESP', 'DTDECLARAC', 'ESCMAEAGR1',
       'STDNEPIDEM', 'STDNNOVA', 'CODPAISRES', 'TPROBSON', 'PARIDADE',
       'KOTELCHUCK', 'CONTADOR', 'munResStatus', 'munResTipo', 'munResNome',
       'munResUf', 'munResLat', 'munResLon', 'munResAlt', 'munResArea'],
      dtype='object')

# Relatório de Políticas Públicas: Análise de Nascimentos em Rondônia (2019)

## Objetivo
Este relatório visa identificar padrões demográficos relacionados à idade dos pais e mães, pesos dos bebês, e distribuição de nascimentos nos municípios de Rondônia, com foco em oferecer dados que possam embasar decisões estratégicas em saúde pública.

## 1. Idade Média das Mães e Pais por Município
A análise revelou que a idade média das mães varia entre 24 e 27 anos, enquanto a idade média dos pais oscila entre 29 e 34 anos. Em alguns municípios, como Cacaulândia e Cabixi, observamos pais com idades médias superiores a 34 anos, o que pode indicar uma tendência local para paternidades mais tardias.

### Recomendações
- Programas voltados ao planejamento familiar e saúde reprodutiva podem focar em orientar pais de faixas etárias mais elevadas, especialmente em municípios onde a idade média dos pais é maior.

## 2. Peso Médio dos Bebês Nascidos no Aniversário do Aluno (05/10/2019)
Foram analisados os pesos médios dos bebês por sexo e escolaridade da mãe. Observa-se uma correlação entre escolaridade materna e peso do bebê, especialmente no caso de mães com escolaridade de 4 a 7 anos e mais de 12 anos, cujos filhos apresentaram pesos médios mais altos.

### Recomendações
- Políticas de incentivo à educação das mulheres podem ser fundamentais para melhorar os resultados de saúde neonatal.
- Programas específicos de nutrição para gestantes com baixa escolaridade também podem ser uma estratégia eficaz para prevenir baixo peso ao nascer.

## 3. Município com Menor Número de Nascimentos em 2019
O município com menor número de nascimentos foi "Município ignorado - RO". As mães nesse município tinham uma idade média de 24 anos e os pais uma média de 22 anos.

### Recomendações
- Considerando a baixa quantidade de nascimentos, investigações mais profundas sobre as condições de acesso à saúde e suporte familiar nessa região podem ser cruciais. Isso pode incluir melhorias no acesso a serviços de saúde para mães e gestantes.

## 4. Município com Maior Número de Nascimentos em Março de 2019
O município com maior número de nascimentos em março foi Porto Velho. A média de filhos vivos por mãe foi de aproximadamente 1.3, com um máximo de 9 filhos vivos, enquanto a idade dos pais variou entre 19 e 62 anos.

### Recomendações
- Dado o elevado número de filhos em algumas famílias, é essencial promover políticas de planejamento familiar.
- Programas de saúde podem ser adaptados para atender a diferentes fases da paternidade, considerando a variação na idade dos pais.

---

Esses resultados podem ser utilizados para criar políticas públicas voltadas à melhoria da saúde materno-infantil, com foco em promover igualdade no acesso a cuidados e orientações para diferentes perfis familiares.
